docs/glossary.md

```diff
@@ -1,9 +1,27 @@
 # Glossary
-## Index
-* RootViewController
-* UI Component
-## RootViewController
+* [General Terms](#general-terms)
+  * [Component](#component)
+  * [Form](#form)
+  * [Response](#response)
+  * [Specification](#specification)
+  
+* [Development](#development)
+  * [RootViewController](#rootviewcontroller)
+  * [UI Component](#ui-component)
+  
+## General Terms
+### Component
+Components are the building block of our forms. They are composed of title, description, requirement and an specification. Date pickers, text fields and selection fields are all components.
+### Form
+The form generated by our framework and composed of title, components and a submit button.
+### Response
+The data provided by the user in the form.
+### Specification
+All components have a specification responsible for providing the components type and its details. Things like the options for a selection component or placeholder text for a text component are specified in the specifications.
+
+## Development
+### RootViewController
 A RootViewController is a UIViewController responsible for being the container of all the UI Components of a given form. This object is the one responsible for the instantiation of each UI Component and for receiving and sending data to the "outside world".
-## UI Component
+### UI Component
```

Comment:
We need to revise our Glossary, especially this earlier definitions of a component.

Reply:
I'm gonna keep the glossary issue open then